src/Build/Evaluation/LazyItemEvaluator.UpdateOperation.cs

```diff
@@ -15,6 +15,9 @@ internal partial class LazyItemEvaluator<P, I, M, D>
         class UpdateOperation : LazyItemOperation
         {
             private readonly ImmutableList<ProjectMetadataElement> _metadata;
+            private ImmutableList<ItemBatchingContext>.Builder _itemsToUpdate = null;
+            private ItemSpecMatchesItem _matchItemSpec = null;
+            private bool? _needToExpandMetadataForEachItem = null;
```

Comment:
Double checking my own review of this file, it looks like the only changes made here are caching the items needed to update and reorganizing some code into functions?

Reply:
Also: +10 points for general code cleanup of legacy code.

Reply:
Yep, pretty much. I should note that I didn't do the caching part because of expected perf gains but because I wanted to access these from different functions.